packages/react/src/components/Shimmer/Shimmer.styles.ts

```diff
@@ -66,7 +66,7 @@ export function getStyles(props: IShimmerStyleProps): IShimmerStyles {
         backgroundColor: shimmerColor || semanticColors.disabledBackground,
         transition: `opacity ${transitionAnimationInterval}ms`,
         selectors: {
-          '> *': {
+          '> div:last-child': {
```

Comment:
This isn't equivalent as the previous selector will match all children of `shimmerWrapper` while this only matches the last child that is also a div. This works fine in the [default case where `ShimmerElementsGroup`](https://github.com/microsoft/fluentui/blob/master/packages/react/src/components/Shimmer/Shimmer.base.tsx#L74-L82) is rendered but a [`customElementsGroup`](https://github.com/microsoft/fluentui/blob/master/packages/react/src/components/Shimmer/Shimmer.types.ts#L30) is not necessarily a `div`.

In practice this change probably won't break anyone but the possibility still exists. Do you have any numbers to show how this change improves performance?

Reply:
I see, I had checked all instances of `customElementsGroup` in the codebase to make sure that the selector still worked but I forgot that users could pass in their own elements too. 

Regarding performance numbers, it's difficult to say exactly how much the improvement would be in time-related measurements because there's so many other factors at play, but these numbers are from a performance profile I collected before (left column)/after (right column) the change, and the amount of work the browser engine needs to do is reduced by 2/3 (match attempts), and is reflected by the total elapsed time as well: 
![image](https://user-images.githubusercontent.com/33800628/176041255-a35d7fe2-9a21-4b4c-8234-e27c0a327b30.png)

Number of matches decreased because I noticed that the selector was also being applied to the div with the `shimmerGradient` classname but it was always being overridden by its own styling, so I thought it was unnecessary to apply it to those elements.

Reply:
Sorry I didn't see your post from yesterday!

Thanks for providing the numbers, this definitely looks like a nice improvement. However, I don't think we can merge it as-is due to the breaking nature of the change. What we've done in the past in situations like this is add a new prop that allows users to opt in to the new, breaking behavior.

Can you add a new prop that controls whether this behavior is enabled or not so users can opt in to the new behavior?

Reply:
That makes sense! Would it be acceptable to have the prop default to true if users don't pass in a `customElementsGroup`? 

And besides adding the prop to the component itself, is there anywhere else I need to make changes? i.e. updating documentation somewhere, creating a new change file?

Reply:
> Would it be acceptable to have the prop default to true if users don't pass in a customElementsGroup?

I like this idea because it will give most people the best perf right out of the box. I'm a little concerned that it will make the implementation a bit more complicated and that will be reflected in the docs because it won't necessarily be straightforward to explain the behavior.

All that said, let's go with this approach and see what it looks like!

> And besides adding the prop to the component itself, is there anywhere else I need to make changes? i.e. updating documentation somewhere, creating a new change file?

Docs are generated from the doc comments on props so just make sure you have a good explanation there. You can verify the docs are updated by:

1. From repo root run `yarn start`
2. Select `@fluentui/public-docsite`
    1. If you get errors try running `yarn buildto @fluentui/public-docsite` then go back to step 1.

> creating a new change file?

We should update the description in the change file as that's what appears in the changelog but let's get the change into the PR first to make sure we're feeling good about it.

Reply:
Got it, thanks!